packages/database/src/core/RepoInfo.ts

```diff
@@ -42,15 +42,20 @@ export class RepoInfo {
     public secure: boolean,
     public namespace: string,
     public webSocketOnly: boolean,
-    public persistenceKey: string = ''
+    public persistenceKey: string = '',
+    private withQueryParam: boolean = false
```

Comment:
`withQueryParam` is hard to interpret out of context.  Consider `includeNamespaceInQueryParams` or something?

Alternatively I wonder if we really need this...  We could probably change `needsQueryParam()` to include `this.host.indexOf(this.namespace + '.') !== 0` in its checks.

Reply:
I don't think this is what we want. Somewhat unfortunately, the database emulator treats the first subdomain as the namespace name only if the host has 3 path components, and otherwise defaults to the value of the `ns` query parameter, if it exists. 

If, for whatever reason, the host url only contained 2 components, and we didn't pass the ns query parameter, we'd get a "missing namespace" response.

The database emulator does prefer the `ns` query parameter in all cases, if it exists. If the semantics we want are to prefer the `ns` query parameter, I think we should not add more logic that depends on the format of the host url and instead do the thing that is simplest to reason about.

Reply:
Bleh.  I don't really know why we have all this code that treats 3 path components specially.  It seems silly.  But rather than deal with potentially breaking stuff, this is fine as-is.

Part of me is tempted to just always include ?ns=blah in the URL as this seems non-harmful (other than a _tiny_ bit of wasted bandwidth) and would mean we don't need all this funky logic.

But it's probably safer to keep this as-is.  Hopefully we don't have to touch this code again.

Reply:
I think we should explore this in later PRs.